modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/_diffs/plugins/tabletoolbar/plugin.js

```diff
@@ -0,0 +1,1475 @@
+					newCell.insertBefore(new CKEDITOR.dom.element(candidateCell));
+					break;
+				} else {
+					candidateCell = null;
+				}
+			}
+
+			// The destination row is empty, append at will.
+
+			if (!candidateCell) {
+				newCellTr.append(newCell);
+			}
+		} else {
+			newCellRowSpan = newRowSpan = 1;
+
+			const newCellTr = tr.clone();
+			newCellTr.insertAfter(tr);
+			newCellTr.append((newCell = cell.clone()));
+
+			const cellsInSameRow = cellInRow(map, rowIndex);
+			for (let i = 0; i < cellsInSameRow.length; i++) {
+				cellsInSameRow[i].rowSpan++;
+			}
+		}
+
+		newCell.appendBogus();
+
+		cell.$.rowSpan = newRowSpan;
+		newCell.$.rowSpan = newCellRowSpan;
+		if (newRowSpan == 1) {
+			cell.removeAttribute("rowSpan");
+		}
+		if (newCellRowSpan == 1) {
+			newCell.removeAttribute("rowSpan");
+		}
+
+		return newCell;
+	}
+
+	function horizontalSplitCell(selection, isDetect) {
+		const cells = getSelectedCells(selection);
+		if (cells.length > 1) {
+			return false;
+		} else if (isDetect) {
+			return true;
+		}
+
+		const cell = cells[0];
+
+		const tr = cell.getParent();
+
+		const table = tr.getAscendant("table");
+
+		const map = CKEDITOR.tools.buildTableMap(table);
+
+		const rowIndex = tr.$.rowIndex;
+
+		const colIndex = cellInRow(map, rowIndex, cell);
+
+		const colSpan = cell.$.colSpan;
+
+		let newColSpan;
+
+		let newCellColSpan;
+
+		if (colSpan > 1) {
+			newColSpan = Math.ceil(colSpan / 2);
+			newCellColSpan = Math.floor(colSpan / 2);
+		} else {
+			newCellColSpan = newColSpan = 1;
+			const cellsInSameCol = cellInCol(map, colIndex);
+			for (let i = 0; i < cellsInSameCol.length; i++) {
+				cellsInSameCol[i].colSpan++;
+			}
+		}
+		const newCell = cell.clone();
+		newCell.insertAfter(cell);
+		newCell.appendBogus();
+
+		cell.$.colSpan = newColSpan;
+		newCell.$.colSpan = newCellColSpan;
+		if (newColSpan == 1) {
+			cell.removeAttribute("colSpan");
+		}
+		if (newCellColSpan == 1) {
+			newCell.removeAttribute("colSpan");
+		}
+
+		return newCell;
+	}
+
+	CKEDITOR.plugins.add("tabletoolbar", {
+		requires: "panelbutton,floatpanel",
+		icons: "add-row, add-column, add-cell",
+		init: function (editor) {
+			function createDef(def) {
+				return CKEDITOR.tools.extend(def || {}, {
+					contextSensitive: 1,
+					refresh(editor, path) {
+						this.setState(
+							path.contains({ td: 1, th: 1 }, 1)
+								? CKEDITOR.TRISTATE_OFF
+								: CKEDITOR.TRISTATE_DISABLED
+						);
+					},
+				});
+			}
+
+			function addCmd(config, def, clickFn) {
+				var commandName = config.commandName || "";
+				var editorFocus = config.editorFocus || 0;
+				var name = config.name;
+				var label = config.label || "";
+				var title = config.title || "";
+				var requiredContent = config.requiredContent || "";
+				var panel = config.panel || {};
+
+				var cmd = editor.getCommand(commandName);
+
+				if (cmd) {
+					return;
+				}
+
+				cmd = editor.addCommand(commandName, def);
+				editor.addFeature(cmd);
```

Comment:
Minor SF: Generally, there should be an empty line between two code instructions. An exception is when multiple lines do the same thing, serve the same purpose. For example, the eight variable declarations above are fine.

Reply:
Makes sense. Changed ;)